File: 合并区间.go
func merge(intervals [][]int) [][]int {
	sort.Slice(intervals,func(i,j int)bool {
		return intervals[i][0] < intervals[j][0]
	})
	
	res := [][]int{}
	for i:=0; i < len(intervals); i++ {
		if len(res) == 0 || intervals[i][0] > res[len(res)-1][1] {
			append(res, intervals[i])
		} else {
			res[len(res)-1][1]=max(res[len(res)-1][1],intervals[i][1])
		}
	}
	return res

}

func max(x,y int)int {
	if x > y {
		return x
	}
	return y
}

File: 合并区间.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
  var sortedIntervals = intervals.sort((a, b) => {
    return a[0] - b[0];
  });
  var merged = [];
  // 遍历数组
  for (var i = 0; i < sortedIntervals.length; i++) {
    if (merged.length === 0 || sortedIntervals[i][0] > merged[merged.length-1][1]) {
      // 推入数组
      merged.push([sortedIntervals[i][0], sortedIntervals[i][1]]);
    } else {
      // 合并区间
      merged[merged.length - 1][1] = Math.max(
        merged[merged.length - 1][1],
        sortedIntervals[i][1],
      );
    }
  }
  return merged;
};
